src/components/Article/NewCommentForm.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useContext } from "react";
+import { UserContext } from "../../contexts/User";
+import { postComment } from "../../../utils/api";
+
+function NewCommentForm({ setComments, article_id }) {
+  const [newComment, setNewComment] = useState("");
+  const [message, setMessage] = useState("");
+  const [isPosting, setIsPosting] = useState(false);
+  const { user } = useContext(UserContext);
+
+  const submitComment = (event) => {
+    event.preventDefault();
+    if (!user) {
+      setMessage("You must be logged in to post a comment");
+    } else if (newComment.length < 5 && newComment.length <= 500) {
+      setMessage("Comments must be between 5 and 500 characters long");
+    } else {
+      setIsPosting(true);
+      setMessage("")
+      postComment(article_id, user, newComment)
+        .then((comment) => {
+          setComments((currComments) => {
+            return [comment, ...currComments];
+          });
+          setMessage("Posted!");
+          setNewComment("");
+          setIsPosting(false);
```

Comment:
This part isn't optimistically rendered as it's waiting for a response from the api call in order to show the comment.

Ideally we would want to show the comment straight away to the user and then make the api call. If the api call works then all is good but if it fails we'll want to inform the user and remove the comment from display.